### PR TITLE
DERDAT 1s duration option

### PR DESCRIPTION
**Pull request by Cynthujah**
1. 1s input option is added in "shiny.R"
2.  1s option is added in "flag_duration_for_updating_if_value_non_standard_and_calced_interval_is_5s" function in "interface.R"

### CODE REVIEW

File: ideal_response/ideal_response_functions.R [hunk 1]
```diff
@@ -72,6 +72,7 @@ norm_p_over_frequency <- function(f, f_ulco, f_upper) {
 }
 
 down_sample_1s <- function(ideal_response_1_s, duration, offset) {
+  if (duration != 1) {   # down-sampling is ignored for 1s data
   ideal_response_1_s <- thicken(
```
Comment (cchristiansen) [inline ideal_response/ideal_response_functions.R:76]: As an if statement is added here, we want to indent the following lines by two spaces.

File: ideal_response/ideal_response_functions.R [hunk 2]
```diff
@@ -96,6 +97,12 @@ down_sample_1s <- function(ideal_response_1_s, duration, offset) {
     group_by(time_group) %>%
     summarise(f = last(f), norm_power = mean(norm_power)) %>%
     as.data.frame()
+  
+  } else if (duration == 1) {
+    ideal_response_downsampled <- ideal_response_1_s
+    colnames(ideal_response_downsampled)[1] <- "time_group"  # only column name is changes from the ideal response (for 1s data analysis)
```
Comment (cchristiansen) [inline ideal_response/ideal_response_functions.R:103]: Typo: "changes" should be "changed".

**Review by cchristiansen — COMMENTED** [review]
Hi Cynthujah, just a couple of minor comments. Otherwise, looks good!

File: ideal_response/ideal_response_functions.R (hunk 2 above)
Comment (Cynthujah) [reply ideal_response/ideal_response_functions.R:103]: Thanks Chirstian. Corrected the typo.

File: docs/documentation_tab.R [hunk 3]
```diff
@@ -324,7 +329,11 @@ documentation_panel <- function() {
       after updating plots on the main tab, in order to save the manually assigned categories to the manual_compliance
       column in the circuit summary dataset, the update plots step needs to be run again on the main tab. The circuits
       are ordered randomly but with a consistent seed value, so the order will remain consistent."
-    )
+    ),
+    h4("Ideal Reponse Curve"),
+    div("'Ideal Response Curve (Ideal reconnection profile)' illustrates changes over dutation. Potential modifications
```
[the remark this answers is not in the document]
Comment (Cynthujah) [reply docs/documentation_tab.R:334]: Thanks Christian. Corrected the typo.

File: ideal_response/ideal_response_functions.R (hunk 1 above)
Comment (Cynthujah) [reply ideal_response/ideal_response_functions.R:76]: Spacing after "if" statement is changed. Thank you.

Comment (phoebeheywood) [reply ideal_response/ideal_response_functions.R]: The arguments for "thicken" should be indented by an additional two spaces for readability

**Review by phoebeheywood — COMMENTED** [review]
Nice work on this Cynthujah, just a few minor changes needed

File: reconnect_compliance/create_reconnection_summary.R [hunk 4]
```diff
@@ -1,22 +1,55 @@
 create_reconnection_summary <- function(combined_data_f,
+                                        duration,
```
[the remark this answers is not in the document]
Comment (Cynthujah) [reply reconnect_compliance/create_reconnection_summary.R:2]: Thank you for the review Pheobe. Added "duration" to the function "create_reconnection_summary". Missing "duration" in the test file "test_create_reconnection_summary.R" is added. Hence the test passes. Also created a separate script for downsampling which is "downsample_for_reconnection.R". It is called in "create_reconnection_summary.R".

File: island_assessment/island_assessment_functions.R [hunk 5]
```diff
@@ -63,10 +63,10 @@ assess_islands <- function(event_window_data) {
       group_by(c_id, clean) %>%
       summarise(
         response_category = first(response_category),
-        max_f = max(fmax),
-        min_f = min(fmin),
-        max_v = (max(vmax) - 240) / 240,
-        min_v = (240 - min(vmin)) / 240
+        max_f = max(fmax,na.rm=TRUE),
```
[the remark this answers is not in the document]
Comment (Cynthujah) [reply island_assessment/island_assessment_functions.R]: Followed the correct style for spacing in both "island_assessment_function.R" and "idel_response_function.R". Thank you.

File: docs/documentation_tab.R [hunk 6]
```diff
@@ -140,6 +140,11 @@ documentation_panel <- function() {
         "Categorised as NA if the circuit does not disconnect/drop to zero during the user specified event window."
       )
     ),
+    h4("Minor discrepancies in the definition of 'Reconnection compliance status' "),
+    div(
+      "Minor discrepancies were observed in the algorithm used for the definition of 'Reconnection compliance status'.
```
[the remark this answers is not in the document]
Comment (Cynthujah) [reply docs/documentation_tab.R]: I added a detailed description. Kindly check. Thank you.

File: ideal_response/ideal_response_functions.R (hunk 1 above)
Comment (Cynthujah) [reply ideal_response/ideal_response_functions.R]: Indented the arguments "thicken" by additional two spaces. Thank you.

File: docs/documentation_tab.R (hunk 6 above)
[the remark this answers is not in the document]
Comment (phoebeheywood) [reply docs/documentation_tab.R]: Looks good, thanks

**Review by phoebeheywood — APPROVED** [review]

**Pull request merged**